bottlecap/src/lifecycle/invocation/context.rs

```diff
@@ -93,8 +93,14 @@ impl ContextBuffer {
             .find(|context| context.request_id == *request_id)
     }
 
-    /// Adds the init duration to a `Context` in the buffer. If the `Context` is not found, a new
-    /// `Context` is created and added to the buffer.
+    /// Creates a new `Context` and adds it to the buffer.
+    ///
+    #[allow(clippy::ptr_arg)]
```

Comment:
Why do we need this?

Reply:
I could change it to &str instead of &String, but I wasn't sure if it is better to keep it consistent with the other functions using &String

```
pub fn create_context(&mut self, request_id: &str) {
        self.insert(Context::new(request_id.to_owned(), 0.0, 0.0, 0, None));
 }
```

Reply:
I just don't think it's worth passing a reference if we are gonna clone inside

Reply:
It doesn't have to be consistent because in the other ones you're searching, and in this one you're keeping the value.

Reply:
I see, thanks!